src/mds/MDCache.cc

```diff
@@ -148,6 +150,11 @@ MDCache::MDCache(MDSRank *m, PurgeQueue &purge_queue_) :
                         (0.9 *(g_conf()->osd_max_write_size << 20));
 
   cache_inode_limit = g_conf().get_val<int64_t>("mds_cache_size");
+
+  // RSS usage has been experimentally shown to be around 25 percent more than the cache_memory_limit. Hence, the cache_memory_limit is reset as per the mds_memory_target(which is an upper bound on the rss usage)
+  uint64_t temp_cache_limit = (uint64_t) (mds->get_mds_memory_target()/1.25);
+  mds->cct->_conf.set_val_default("mds_cache_memory_limit", stringify(temp_cache_limit));
```

Comment:
I know I've probably asked you to move this code here but consider that the value of `mds_memory_target` -- whether set manually or the default -- always overrides the default for `mds_cache_memory_limit`. We should update the default for `mds_cache_memory_limit` to reflect what its value would be if `mds_memory_target`'s default value is used. IOW, with this change, `mds_cache_memory_limit`'s 1GB default value is never used.

Reply:
Okay, sorry I've been flip flopping on this. I think the right thing to do is not change the default of the `mds_cache_memory_limit`. We can discard this modification based on the "25% more..." observation. The `MDCache::resize` method, if it works properly, would eventually arrive at that value itself without help.

We also don't want to get in the business of modifying the default of a config option. Consider an admin trying to play with the values of `mds_cache_memory_limit` and `mds_memory_target`. It's confusing how they interact and it's not possible to "undo" setting them such that the defaults are used again.

Reply:
Ack. I agree It makes much more sense to let the resize do its work.